src/scanpy/preprocessing/_simple.py

```diff
@@ -715,14 +756,27 @@ def regress_out(
             regres = regressors
         tasks.append(tuple((data_chunk, regres, variable_is_categorical)))
 
-    from joblib import Parallel, delayed
+    flag = None
+    if not variable_is_categorical:
+        A = regres.to_numpy()
+        # checking if inverse of A.T@A matix is zero or not.
```

Comment:
That comment doesn’t add anything that the code doesn’t say. Can you replace it with a comment explaining *why* you do that?